docs/contributing.md

```diff
@@ -103,6 +103,44 @@ Follow these steps to contribute code:
    Consult [GitHub Help](https://help.github.com/articles/about-pull-requests/)
    for more information on using pull requests.
 
+
+## Troubleshooting
+
+### Too many commits in a PR
+
+If your PR has too many commits associated with it, then our build process will
+fail with an error message. This is because of two reasons:
+
+* We prefer to keep our commit history clean.
+
+* Our source sync process will actually fail if our commit tree is too large.
+
+If you encounter this error message, you should squash your commits. You can do
+this in various ways, and we demonstrate two options here.
+
+* If you simply want to squash the last 8 commits and write a new commit
+  message from scratch, make sure you checkout your branch and run:
+
+  ```bash
+  git reset --soft HEAD~8 && git commit
+  ```
```

Comment:
The workflow I personally use is `$git rebase master && git reset --soft master && git commit`. Seems safer than counting commits... what do you think?

Reply:
Good idea, updated it, and also updated the explanation!